lib/CssSelectorAbstract.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/**
+ * An alternative abstract class for bridges utilizing CSS Selectors.
+ **/
+abstract class CssSelectorAbstract extends UserDefinedAbstract {
```

Comment:
Did you consider using these classes as helpers, rather than abstract base classes? I usually do not like to promote design patterns indiscriminately but [composition over inheritance](https://en.wikipedia.org/wiki/Composition_over_inheritance) is one of the better ones.

Reply:
Or if ease of use is preferred over conceptual purity, [traits](https://www.php.net/manual/en/language.oop5.traits.php) are a good compromise.